### PR TITLE
updated jwt example to use IfMiddleware

### PR DESCRIPTION
**Pull request by StephanDollberg**
As promised, I have updated the example to use the new IfMiddleware.

### CODE REVIEW

File: jwt/main.go [hunk 1]
```diff
@@ -13,35 +14,32 @@ func handle_auth(w rest.ResponseWriter, r *rest.Request) {
...
-			return false
+			return userId == "admin" && password == "admin"
 		}}
 
-	login_api := rest.NewApi()
-	login_api.Use(rest.DefaultDevStack...)
-	login_router, _ := rest.MakeRouter(
+	api := rest.NewApi()
+	// we use the IfMiddleware to remove certain paths from needing authentication
+	api.Use(&rest.IfMiddleware{
```
Comment (ant0ine) [inline jwt/main.go]: IfTrue and IfFalse are optional, you may want to write:

```go
	api.Use(&rest.IfMiddleware{
		Condition: func(request *rest.Request) bool {
			return request.URL.Path != "/login"
		},
		IfTrue: jwt_middleware,
	})
```

File: jwt/main.go [hunk 2]
```diff
@@ -13,35 +14,32 @@ func handle_auth(w rest.ResponseWriter, r *rest.Request) {
...
+		Condition: func(request *rest.Request) bool {
+			return request.URL.Path == "/login"
+		},
+		IfTrue:  nil,
+		IfFalse: jwt_middleware,
+	})
+	api.Use(rest.DefaultDevStack...)
```
Comment (ant0ine) [inline jwt/main.go]: The DefaultDevStack includes error handling and logging. I think it should be moved before `api.Use(&rest.IfMiddleware{`.
That was, auth error will be correctly logged.

**Comment by ant0ine** [comment]
Thanks @StephanDollberg ! This is much nicer than with the two api objects! See my two comments.

**Comment by StephanDollberg** [comment]
Thanks for your remarks.

Concerning the position of the JWT middleware, I didn't think about this and did some tests. What I noticed is that unauthorized requests to endpoints that require auth are not logged independent of whether I put the JWT middleware before or after the DevStack. Any ideas what might be the reason here?

Cheers,
Stephan

**Comment by ant0ine** [comment]
Interesting, in theory, the following line of the JWT middleware:

```go
	rest.Error(writer, "Not Authorized", http.StatusUnauthorized)
```
should be caught be the AccessLogApache Middleware (included in the DevStack) and generate a log line about the failed request.

I'll try to reproduce later today.

**Comment by ant0ine** [comment]
I've just moved the line
```go
        api.Use(rest.DefaultDevStack...)
```
just below
```go
        api := rest.NewApi()
```
and then:
```
curl -i http://localhost:8080/api/auth_test
```
produces the following log line:
```
29/Mar/2015:23:12:16 +0000 401 137μs "GET /auth_test HTTP/1.1" - "curl/7.37.1"
```

**Comment by StephanDollberg** [comment]
Ok, sorry for the waste of time. Worked for me today as well. Probably did something stupid the day before.

I have updated the commit with your remarks now.

Cheers and sorry again,
Stephan

**Comment by ant0ine** [comment]
No worries, Thanks for the update!

**Pull request merged**